### PR TITLE
Avner

### PR DESCRIPTION
**Pull request by avnerlavi**
code snippets of up to chapter 4

### CODE REVIEW

File: numerical_analysis.py [hunk 1]
```diff
@@ -57,8 +57,24 @@ def get_soft_scores_and_true_labels(dataset, model):
...
+    all_first_soft_scores, all_second_soft_scores, gt_labels = [], [], []
+    batch_size = 32
+    dataloader = DataLoader(dataset,
+                            batch_size=batch_size,
+                            shuffle=False)
+    model = model.to(device)
+
+    with torch.no_grad():
+        for batch_idx, (inputs, targets) in enumerate(dataloader):
+            inputs, targets = inputs.to(device), targets
```
Comment (lev1ayz) [inline numerical_analysis.py:69]: Why "targets = targets"?

File: numerical_analysis.py [hunk 2]
```diff
@@ -57,8 +57,24 @@ def get_soft_scores_and_true_labels(dataset, model):
...
         gt_labels: an iterable holding the samples' ground truth labels.
     """
-    """INSERT YOUR CODE HERE, overrun return."""
-    return torch.rand(100, ), torch.rand(100, ), torch.randint(0, 2, (100, ))
+    all_first_soft_scores, all_second_soft_scores, gt_labels = [], [], []
+    batch_size = 32
+    dataloader = DataLoader(dataset,
+                            batch_size=batch_size,
+                            shuffle=False)
+    model = model.to(device)
```
Comment (lev1ayz) [inline numerical_analysis.py:65]: I think the model should also be put in eval mode, unless it's called before the function

File: trainer.py [hunk 3]
```diff
@@ -60,7 +60,20 @@ def train_one_epoch(self) -> tuple[float, float]:
         print_every = int(len(train_dataloader) / 10)
```
Comment (lev1ayz) [inline trainer.py:61]: should probably add self.model.train()

File: trainer.py [hunk 4]
```diff
@@ -91,12 +104,23 @@ def evaluate_model_on_dataloader(
         correct_labeled_samples = 0
         print_every = max(int(len(dataloader) / 10), 1)
```
Comment (lev1ayz) [inline trainer.py:106]: should probably add model.eval()

**Review by lev1ayz — COMMENTED** [review]
Looks good, i didn't have a lot to add, let me know what you think and then we'll merge it

File: numerical_analysis.py (hunk 1 above)
Comment (avnerlavi) [reply numerical_analysis.py:69]: originally tried targets.to(device), turned out it wasn't needed  but forgot to remove thts bit- good catch

File: trainer.py (hunk 3 above)
Comment (avnerlavi) [reply trainer.py:61]: called in start of function

File: trainer.py (hunk 4 above)
Comment (avnerlavi) [reply trainer.py:106]: called in start of function

File: numerical_analysis.py (hunk 2 above)
Comment (avnerlavi) [reply numerical_analysis.py:65]: it's set in the main() function, assuming we're not going to call the get_soft_scores_and_true_labels() function outside of it I think we're good
